modules/xml/src/smithy4s/xml/internals/XmlSchemaVisitor.scala

```diff
@@ -0,0 +1,138 @@
+      member: Schema[A]
+  ): XmlDecoder[C[A]] = {
+    val xmlName = member.hints.get(XmlName).map(_.value).getOrElse("member")
+    val isFlattened = hints.has(XmlFlattened)
+    val memberReader = compile(member)
+    new XmlDecoder[C[A]] {
+      def read(cursor: XmlCursor): Either[XmlReadError, C[A]] = {
+        val realCursor = if (isFlattened) cursor else cursor.down(xmlName)
+        realCursor match {
+          case XmlCursor.MultipleNodes(history, nodes) =>
+            nodes.zipWithIndex
+              .traverse { case (elem, index) =>
+                memberReader.read(
+                  XmlCursor.SingleNode(history.appendIndex(index), elem)
+                )
+              }
+              .map(list => tag.fromIterator(list.iterator))
+          case XmlCursor.SingleNode(history, elem) =>
+            memberReader
+              .read(XmlCursor.SingleNode(history.appendIndex(0), elem))
+              .map(value => tag.fromIterator(Iterator.single(value)))
+          case XmlCursor.NoNode(_) => Right(tag.empty)
+          case other =>
+            Left(XmlReadError(other.history, s"Expected one or multiple nodes"))
+        }
+      }
+    }
+  }
+
+  def map[K, V](
+      shapeId: ShapeId,
+      hints: Hints,
+      key: Schema[K],
+      value: Schema[V]
+  ): XmlDecoder[Map[K, V]] = ???
+
+  def enumeration[E](
+      shapeId: ShapeId,
+      hints: Hints,
+      values: List[EnumValue[E]],
+      total: E => EnumValue[E]
+  ): XmlDecoder[E] = ???
+
+  def struct[S](
+      shapeId: ShapeId,
+      hints: Hints,
+      fields: Vector[SchemaField[S, _]],
+      make: IndexedSeq[Any] => S
+  ): XmlDecoder[S] = {
+    def fieldReader[A](field: SchemaField[S, A]): XmlDecoder[A] = {
+      val isAttribute = field.instance.hints.has(XmlAttribute)
+      val xmlName =
+        field.instance.hints.get(XmlName).map(_.value).getOrElse(field.label)
+      field
+        .foldK(new Field.FolderK[Schema, S, XmlDecoder] {
+          def onRequired[AA](
+              label: String,
+              instance: Schema[AA],
+              get: S => AA
+          ): XmlDecoder[AA] = {
+            if (isAttribute) compile(instance).attribute(xmlName)
+            else compile(instance).down(xmlName)
+          }
+          def onOptional[AA](
+              label: String,
+              instance: Schema[AA],
+              get: S => Option[AA]
+          ): XmlDecoder[Option[AA]] = {
+            if (isAttribute) compile(instance).optional.attribute(xmlName)
+            else compile(instance).optional.down(xmlName)
+          }
+        })
+    }
+    val readers = fields.map(fieldReader(_))
+    new XmlDecoder[S] {
+      def read(cursor: XmlCursor): Either[XmlReadError, S] =
+        readers.traverse(_.read(cursor)).map(make)
+    }
+  }
+
+  def union[U](
+      shapeId: ShapeId,
+      hints: Hints,
+      alternatives: Vector[SchemaAlt[U, _]],
+      dispatch: Alt.Dispatcher[Schema, U]
+  ): XmlDecoder[U] = ???
```

Comment:
why does CI not complain about these

![image](https://user-images.githubusercontent.com/894884/194282578-8e92ff4d-7466-4ec5-8d67-cab241bf9008.png)

oh I guess it's because you didn't aggregate the xml module in root

Reply:
even so CI wouldn't complain about them I think, because it's not deadcode.